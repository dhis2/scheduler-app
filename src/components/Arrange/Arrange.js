import React from 'react'
import { node, bool } from '@dhis2/prop-types'
import styles from './Arrange.module.css'

/**
 * Arranges children either horizontally or vertically. Arranges horizontally by default.
 */

const Arrange = ({ children, vertical }) => {
    const className = vertical ? styles.vertical : styles.horizontal

    return <div className={className}>{children}</div>
}

Arrange.defaultProps = {
    vertical: false,
}

Arrange.propTypes = {
    children: node.isRequired,
    vertical: bool,
}

export default Arrange
