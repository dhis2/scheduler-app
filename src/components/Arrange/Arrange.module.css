.horizontal,
.vertical {
    display: flex;
    align-items: center;
}

.vertical {
    flex-direction: column;
}
