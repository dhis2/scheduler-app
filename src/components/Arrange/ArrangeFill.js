import React from 'react'
import { node } from 'prop-types'
import styles from './ArrangeFill.module.css'

/**
 * Fills all available space
 */

const ArrangeFill = ({ children }) => {
    return <div className={styles.fill}>{children}</div>
}

ArrangeFill.propTypes = {
    children: node.isRequired,
}

export default ArrangeFill
