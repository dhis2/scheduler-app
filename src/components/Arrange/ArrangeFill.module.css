.fill {
    flex-grow: 1;
}
