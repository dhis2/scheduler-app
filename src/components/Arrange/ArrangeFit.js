import React from 'react'
import { node } from '@dhis2/prop-types'
import styles from './ArrangeFit.module.css'

/**
 * Shrink container to fit contents
 */

const ArrangeFit = ({ children }) => {
    return <div className={styles.fit}>{children}</div>
}

ArrangeFit.propTypes = {
    children: node.isRequired,
}

export default ArrangeFit
