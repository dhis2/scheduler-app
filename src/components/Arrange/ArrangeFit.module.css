.fit {
    flex-grow: 0;
}
