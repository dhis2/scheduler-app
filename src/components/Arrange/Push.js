import React from 'react'
import { node, oneOf } from 'prop-types'
import styles from './Push.module.css'

const Push = ({ children, direction }) => (
    <div className={styles[direction]}>{children}</div>
)

Push.propTypes = {
    children: node.isRequired,
    direction: oneOf(['up', 'down', 'left', 'right']).isRequired,
}

export default Push
