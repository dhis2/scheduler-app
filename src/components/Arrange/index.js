import Arrange from './Arrange'
import Push from './Push'

Arrange.Push = Push

export { Arrange }
