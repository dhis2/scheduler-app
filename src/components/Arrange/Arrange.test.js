import React from 'react'
import { shallow } from 'enzyme'
import Arrange from './Arrange'

describe('<Arrange>', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<Arrange>Text</Arrange>)

        expect(wrapper).toMatchSnapshot()
    })

    it('responds to a vertical prop', () => {
        const wrapper = shallow(<Arrange vertical>Text</Arrange>)

        expect(wrapper).toMatchSnapshot()
    })
})
