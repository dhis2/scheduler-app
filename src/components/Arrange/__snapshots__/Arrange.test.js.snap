// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Arrange> renders correctly 1`] = `
<div
  className="horizontal"
>
  Text
</div>
`;

exports[`<Arrange> responds to a vertical prop 1`] = `
<div
  className="vertical"
>
  Text
</div>
`;
